funcframework/framework.go

```diff
@@ -206,6 +212,64 @@ func wrapEventFunction(fn interface{}) (http.Handler, error) {
 	}), nil
 }
 
+func wrapTypedFunction(fn interface{}) (http.Handler, error) {
+	err := validateTypedFunction(fn)
+	if err != nil {
+		return nil, err
+	}
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, err := readHTTPRequestBody(r)
+		if err != nil {
+			writeHTTPErrorResponse(w, http.StatusBadRequest, crashStatus, fmt.Sprintf("%v", err))
+			return
+		}
+		argVal := reflect.New(reflect.TypeOf(fn).In(0))
+
+		if err := json.Unmarshal(body, argVal.Interface()); err != nil {
+			writeHTTPErrorResponse(w, http.StatusBadRequest, crashStatus, fmt.Sprintf("Error: %s, while converting input type data: %s", err.Error(), string(body)))
+			return
+		}
+
+		defer recoverPanic(w, "user function execution")
+		funcReturn := reflect.ValueOf(fn).Call([]reflect.Value{
+			argVal.Elem(),
+		})
+
+		if len(funcReturn) >= 1 {
+			var errorVar error
+			errorType := reflect.TypeOf(&errorVar).Elem()
+			firstVal := funcReturn[0].Interface()
+			if !reflect.TypeOf(firstVal).AssignableTo(errorType) {
+				returnVal, _ := json.Marshal(firstVal)
+				fmt.Fprintf(w, string(returnVal))
+			}
+
+			lastVal := funcReturn[len(funcReturn)-1].Interface()
+			if lastVal != nil && reflect.TypeOf(lastVal).AssignableTo(errorType) {
+				writeHTTPErrorResponse(w, http.StatusInternalServerError, errorStatus, fmtFunctionError(lastVal))
+			}
+
+		}
+	}), nil
+}
+
+func validateTypedFunction(fn interface{}) error {
```

Comment:
Can we make this function return the result of `reflect.TYpeOf(fn).In(0)` so that the `wrapTypedFunction` doesn't ahve to redo that bit? 

Checking an 0-index without seeing the if len() > 0 check nearby is kinda scary.